src/pushsource/_impl/model/ami.py

```diff
@@ -99,9 +99,33 @@ class AmiPushItem(VMIPushItem):
     )
     """Billing codes associated with this image."""
 
+    release_notes = attr.ib(type=str, default=None, validator=optional_str)
+    """"Notes regarding changes in the newest version."""
+
+    usage_instructions = attr.ib(type=str, default=None, validator=optional_str)
+    """Step by step instructions for the end-user to launch, configure and access the product. 
+
+    Write with a less-technical customer in mind.
+    """
+
+    recommended_instance_type = attr.ib(type=str, default=None, validator=optional_str)
+    """Recommended instance type for AMI e.g. "t2.micro", "r6a.large" """
+
+    marketplace_entity_type = attr.ib(type=str, default=None, validator=optional_str)
+    """Type of entity e.g. "AMIProduct", "SaasProduct", "ServerProduct". """
+
     image_id = attr.ib(type=str, default=None, validator=optional_str)
     """AMI Image ID used to reference the image in AWS."""
 
+    scanning_port = attr.ib(type=int, default=None, validator=instance_of(int))
+    """AMI scanning port, used when importing the AMI into AWS Marketplace to validate the AMI."""
+
+    user_name = attr.ib(type=str, default=None, validator=optional_str)
+    """The username used to login to the AMI. (Usually set to ec2-user)"""
+
+    security_groups = attr.ib(type=str, default=None, validator=optional_str)
+    """Automatically created security groups for the product. """
```

Comment:
Group(s) (plural) but the type is a string?

If correct, how does the string contain more than one group?  e.g. comma-separated?  This detail should be mentioned.

Reply:
I checked, it is supposed to be some list of strings, which could be converted to a meaningful JSON with some converter. I am applying the fix.

Reply:
This will need to be something similar to BillingCode but it'll need to be a List[SecurityGroup] so that the data ends up looking something like this:

"SecurityGroups": [
                                {
                                    "FromPort": 22,
                                    "IpProtocol": "tcp",
                                    "IpRanges": [
                                        "[0.0.0.0/0](http://0.0.0.0/0)"
                                    ],
                                    "ToPort": 22
                                }
                            ]

Reply:
Implemented.